src/bidiMapper/domains/context/BrowsingContextImpl.ts

```diff
@@ -853,6 +909,94 @@ export class BrowsingContextImpl {
   }
 }
 
+type Rect = {
+  x: number;
+  y: number;
+  width: number;
+  height: number;
+};
+
+function deserializeDOMRect(rect: Script.EvaluateResult) {
```

Comment:
Function does more then it should as it not only serializes but checks how the the command responded.
Also you can save a lot of `reac.` if use destruct on property. 
This can be a helper on the class and create sandbox and other parts.

Reply:
I think it's still "correct" to check how the command responded. After all, it's converting `Script.EvaluateResult`. I've tried to use destructuring but it doesn't work because of type checking.

Reply:
I think the function should accept a Script.RemoveValue (or similar type) rather then a command reponse.